language/diem-framework/modules/0L/Subsidy.move

```diff
@@ -52,22 +53,98 @@ address 0x1 {
+        let node_address = Vector::borrow<address>(outgoing_set, i);
         // Transfer gas from vm address to validator
-        let minted_coins = Diem::mint<GAS>(vm, subsidy_granted);
-        DiemAccount::vm_deposit_with_metadata<GAS>(
-          vm,
-          node_address,
-          minted_coins,
-          b"validator subsidy",
-          b""
-        );
+        check_team_and_pay(vm, node_address, subsidy_granted);
 
         // refund operator tx fees for mining
-        refund_operator_tx_fees(vm, node_address);
+        refund_operator_tx_fees(vm, *node_address);
         i = i + 1;
       };
     }
+    use 0x1::Debug::print;
+
+    fun check_team_and_pay(vm: &signer, captain_address: &address, subsidy_granted: u64) {
+      // this is a solo validator. Exists during transition to delegation mode. This is a fallback condition to keep the node from halting
+      print(&100000);
+      let captain_value = subsidy_granted;
+      print(&subsidy_granted);
+      if (Teams::team_is_init(*captain_address)) {
+        print(&100100);
+        // split captain reward and send to captain.
+        let captain_pct = Teams::get_operator_reward(*captain_address);
+        print(&captain_pct);
+        // split off the captain value
+        captain_value = FixedPoint32::multiply_u64(
+          subsidy_granted,
+          FixedPoint32::create_from_rational(captain_pct, 100) 
+        );
+        print(&100200);
+        print(&captain_value);
+
+        let value_to_members = subsidy_granted - captain_value;
+        print(&value_to_members);
+
+        // get team members
+        let members = Teams::get_team_members(*captain_address);
+        // split the team subsidy
+        print(&100300);
+        split_subsidy_to_team(vm, &members, value_to_members);
+      };
+      print(&100400);
+      print(&captain_value);
+      let captain_coins = Diem::mint<GAS>(vm, captain_value);
+      
+      let pre_balance = DiemAccount::balance<GAS>(*captain_address);
+      print(&pre_balance);
+      
+      // payment to captain
+      DiemAccount::vm_deposit_with_metadata<GAS>(
+        vm,
+        *captain_address,
+        captain_coins,
+        b"validator subsidy",
+        b""
+      );
+    }
+
+    public fun split_subsidy_to_team(vm: &signer, members: &vector<address>, value_to_members: u64) {
+      let collective_height = TowerState::collective_tower_height(members);
+      print(&100310);
+      print(members);
+      print(&collective_height);
+      print(&value_to_members);
+      let i = 0;
+      while (i < Vector::length(members)) {
+        let addr = Vector::borrow(members, i);
+        let one_height = TowerState::tower_for_teams(*addr);
+        print(&100320);
+        print(&one_height);
+        if (one_height > 0) {
+          let payment_to_this_member = FixedPoint32::multiply_u64(
+            value_to_members,
+            FixedPoint32::create_from_rational(one_height, collective_height)
```

Comment:
I have a design idea here.

The min tower height is determined by the number of towers who are a members in a team. Basically as you add members to a team the min height increases. I think this would mitigate sybiks

Reply:
@zmanian is this the min height of a member to join a team?  Is this like musical chairs in a sense, where people get knocked off a team the  hight to join a team get's progressively higher?

Reply:
My mental model is that increasing height requirement on the team can prevent the n+1 team member from join but you don't get kicked out of team if the difficulty becomes higher than your height

Reply:
So then even if they make a new team, the rate of growth for that team would be limited…This idea sounds quite promising

Reply:
So this means that new members to the network need to go to new teams (teams with fewer members). Does this increase the sybil resistance?

Reply:
yes because those new teams will have low aggregate voting power

Reply:
So just to be clear. The sybil case we are concerned about is a consensus fault due to 2/3 validators having made sybil collective towers, yes? If so I don't really see that risk being that high (given how the validator tree grows), and the cost to users is quite high, it's hard to explain to end-user miners what the obvious thing to do. And IMO the obvious thing to do is to pick a team captain that runs reliable infrastructure and is participating in the community. Are we ok with that penalty?

Reply:
Below is a super crude suggestion.

There is a term called equalizer in APA(pool) leagues .
In the league 5 players whose combined ratings should not exceed 21 form a team  and are pitted against another team with a similar structure. (A player is rated on a scale from 2 to 7 . New players start at 2)


One of the reasons why this works well is because it incentivizes teams to find lower rated players  & as players gain  in ratings they leave and join new teams or form their own ones . But the demand for lower rated (new) players always exist. This organically grows into more and more teams being formed over multiple tournaments .

In 0Ls case , players would be  members with towers , and  rating would be akin to tower heights.

The tradeoff  is that we deviate away from an earlier suggested axiom above that no member should be blocked from joining a team . It can be mitigated if we set the member limit to each team to be slightly higher than  "(available members) / (Total Teams). Teams can negotiate exchange of members too if they so choose . 

Does this  sounds promising ?

Reply:
@Vee-Emm This is fascinating. Need to think more about it.
I think the thing to remember is that we want the users to "do the obvious thing", and the obvious thing is to pick the right stewards of the network; join Validators that have an engaged and technical operator.

My proposal is to use @sm86 's idea and do sum of squares to count the collective tower, create an increasing threshold on a per team basis (similar to @zmanian), and increase the rate-limit on account creation (to 4 or 6 weeks)

With these measures I think it's unlikely that one actor can take 1/3rds of consensus seats to halt network, and less likely still to get to 2/3rds and update the state machine.

Reply:
I think the sum of squares approach makes sense (or at least sum of some super-linear function, squares is simple, but maybe not optimal, depending on what exactly we want to achieve). Just to confirm, the rate limit would limit the number of teams that can be created, not the number of people that can join teams (which would be unbounded), is that correct?

Reply:
The rate-limit I refer to is the creation of new validator accounts, as we have today. So effectively new teams. The increase would be from 2 weeks to 4 or 6 weeks. To give ample time to catch attacks coming.

Reply:
Looping back on this discussion. The final implementation uses a threshold of the RMS of the towers. Which achieves the goal of sum of squares per @jamesmeijers

Reply:
Is the RMS of towers threshold specifically for being able to join a team? 

So phase 1 keeps the validator tower height as the metric for validator set inclusion and consensus. Then, in phase 2 of the rollout, will the collective (not RMS) towers of the validator/team be the metric in order to incentivize team formation?

In phase 2, will the collective delay tower difficulty for teams target 100/max validators to limit subsidy minting?

Reply:
@misko9 This is covered in the ol/documentaiton/teams. In short yes the Phase 1 does not change qualification or consensus weights. Phase 2 Does. And there are things to work out about the sybil resistance of phase 2, including how the collective difficulty changes.